backend/src/database/models.rs

```diff
@@ -344,6 +344,56 @@ impl Campaign {
             .get_result(conn)
             .ok()
     }
+
+    pub fn delete(conn: &PgConnection, campaign_id: i32) -> bool {
+        use crate::database::schema::campaigns::dsl::*;
+
+        diesel::delete(campaigns.filter(id.eq(campaign_id)))
+            .execute(conn)
+            .is_ok()
+    }
+
+    // this function is used to delete a campaign and all associated data
+    // ie the campaign, roles, applications, questions, answers, ratings and comments.
+    // it will not delete users from OrganisationUsers
+    pub fn delete_deep(conn: &PgConnection, campaign_id: i32) -> Option<()> {
```

Comment:
Not sure if diesel does anything behind the scenes to mitigate this effect, but I'm a little suss that this is all happening in individual queries when it would be much better batched...

Reply:
I couldnt find anything that would do batching :( However, deleting a campaign fully (as opposed to hiding it from public) is something that should rarely happen so happy to leave it for now